### PR TITLE
[BUGS-5990] Prefix surrogate keys with blog ID on WPMS

### PR DESCRIPTION
**Pull request by jspellman814**
Addresses #196. 

This change will have no affect on single site installs. 

For users who already have this plugin installed on a WPMS, they will need to click the `Clear Cache` button on the settings page (`wp-admin/options-general.php?page=pantheon-cache`) to receive the prefixed keys.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,6 +308,9 @@ See [CONTRIBUTING.md](https://github.com/pantheon-systems/pantheon-advanced-page
 
 ## Changelog ##
 
+## 1.3.0-dev (April 13, 2023) ##
```
Comment (pwtyler) [inline README.md:311]:
```suggestion
## Latest ##
```
Keep this to `Latest` in `develop` and feat branches and change in the release PR.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 **Tags:** pantheon, cdn, cache 
 **Requires at least:** 4.7  
 **Tested up to:** 6.2  
-**Stable tag:** 1.2.3 
+**Stable tag:** 1.3.0-dev
```
Comment (pwtyler) [inline README.md:7]: 🎊

File: readme.txt [hunk 3]
```diff
@@ -302,6 +302,10 @@ Pantheon Advanced Page Cache integrates with WordPress plugins, including:
 See [CONTRIBUTING.md](https://github.com/pantheon-systems/wp-saml-auth/blob/master/CONTRIBUTING.md) for information on contributing.
 
 == Changelog ==
+
+= 1.3.0-dev (April 13, 2023) =
```
Comment (pwtyler) [inline readme.txt:306]:
```suggestion
= Latest =
```
Same as above. Update this in release PR.

File: readme.txt [hunk 4]
```diff
@@ -302,6 +302,10 @@ Pantheon Advanced Page Cache integrates with WordPress plugins, including:
...
+* Adds support for WordPress Multisite [[#228](https://github.com/pantheon-systems/pantheon-advanced-page-cache/pull/228)].
```
Comment (pwtyler) [inline readme.txt:307]: In the PR Body you mentioned

> For users who already have this plugin installed on a WPMS, they will need to click the Clear Cache button on the settings page to receive the prefixed keys.

We might want to make this a little more visible in the .org README's release notes as well?

Comment (jspellman814) [reply readme.txt]: Good point! Updated this note in [1ac079a](https://github.com/pantheon-systems/pantheon-advanced-page-cache/pull/228/commits/1ac079aa93b5265e99660dab719f69154cfa8cf2).

File: README.md [hunk 5]
```diff
@@ -308,6 +308,9 @@ See [CONTRIBUTING.md](https://github.com/pantheon-systems/pantheon-advanced-page
 
 ## Changelog ##
 
+## Latest ##
+* Adds support for WordPress Multisite which resolves issue where editing a Post on one subsite clears the home page cache of other sites in the Multisite install if it has a Post containing the same ID. For users who already have this plugin installed on a WPMS, they will need to click the Clear Cache button on the settings page to generate the prefixed keys [[#228](https://github.com/pantheon-systems/pantheon-advanced-page-cache/pull/228)].
```
Comment (jazzsequence) [inline README.md:312]: The note in the changelog for users (referenced [here](https://github.com/pantheon-systems/pantheon-advanced-page-cache/pull/228#discussion_r1166078296)) should be moved to the `== Upgrade Notice ==` section of the readme so it is more visible to users than just being in the changelog.

File: inc/class-emitter.php [hunk 6]
```diff
@@ -313,6 +314,7 @@ public static function get_rest_api_surrogate_keys() {
 		 */
 		$keys = self::get_instance()->rest_api_surrogate_keys;
 		$keys = array_unique( $keys );
```
Comment (jazzsequence) [inline inc/class-emitter.php:316]: Is this `array_unique` unnecessary since we're doing it again on 319?
🔢

Comment (jspellman814) [reply inc/class-emitter.php:316]: Probably? Can test.

Comment (jspellman814) [reply inc/class-emitter.php]: I've been testing removing this this morning, and haven't come across any issues. Pushed commit [dc0c290](https://github.com/pantheon-systems/pantheon-advanced-page-cache/pull/228/commits/dc0c290c41ad22da9f18f3378c37b52432124fef) removing the duplicate.

File: README.md [hunk 7]
```diff
@@ -362,3 +365,8 @@ See [CONTRIBUTING.md](https://github.com/pantheon-systems/pantheon-advanced-page
 
 ### 0.1.0 (November 23rd, 2016) ###
 * Initial release.
+
+## Upgrade Notice ##
+
+### Latest ###
+**Note that the Pantheon Advanced Page Cache Latest release now prefixes keys on WPMS with the blog ID. For users who already have this plugin installed on a WPMS, they will need to click the Clear Cache button on the settings page to generate the prefixed keys.
```
Comment (pwtyler) [inline README.md:372]: 🔢 
```suggestion
Note that the Pantheon Advanced Page Cache Latest release now prefixes keys on a WordPress Multisite (WPMS) with the blog ID. For users who already have this plugin installed on a WPMS, they will need to click the Clear Cache button on the settings page to generate the prefixed keys.
```

Also, are you going for bullets here with ** or is that meant to be read as an asterisk? I would drop either way but optional.

**Review by CdrMarks — DISMISSED** [review]

**Review by CdrMarks — DISMISSED** [review]

**Review by pwtyler — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -4,7 +4,7 @@
 **Tags:** pantheon, cdn, cache  
 **Requires at least:** 4.7  
 **Tested up to:** 6.2  
-**Stable tag:** 1.2.4  
+**Stable tag:** 1.3.0-dev
```
Comment (pwtyler) [inline README.md:7]: Just learned this— two spaces at the end of line line for the line break to appear properly (preview md to see it now looks weird)
```suggestion
**Stable tag:** 1.3.0-dev  
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -8,17 +8,34 @@ Pull requests and issues are welcome!
...
 
-1. Starting from `develop`, cut a release branch named `release_X.Y.Z` containing your changes.
-1. Update plugin version in `package.json`, `README.md`, `readme.txt`, and `pantheon-advanced-page-cache.php`.
-1. Update the Changelog with the latest changes.
-1. Create a PR against the `master` branch.
-1. After all tests pass and you have received approval from a CODEOWNER (including resolving any merge conflicts), merge the PR into `master`.
-1. Pull `master` locally, create a new tag, and push up.
+1. From `develop`, checkout a new branch `release_X.Y.Z`.
+1. Make a release commit:
+    * Drop the `-dev` from the version number in [readme.txt](./readme.txt) and [wp-decoupled-preview.php](./wp-decoupled-preview.php)
```
Comment (pwtyler) [inline CONTRIBUTING.md:23]: Fix files referenced
```suggestion
    * Drop the `-dev` from the version number in `package.json`, `README.md`, `readme.txt`, and `pantheon-advanced-page-cache.php`.
```

**Review by pwtyler — APPROVED** [review]

**Pull request merged**